Source/WebCore/style/StyleBuilder.cpp

```diff
@@ -92,6 +92,9 @@ void Builder::applyAllProperties()
 // High priority properties may affect resolution of other properties (they are mostly font related).
 void Builder::applyHighPriorityProperties()
 {
+    // writing-mode defines vi/vb units which may be used in other high priority properties
+    applyProperties(CSSPropertyWritingMode, CSSPropertyWritingMode);
+
     applyProperties(CSSPropertyWebkitRubyPosition, CSSPropertyWebkitRubyPosition);
```

Comment:
It would be nice if these were applied based on the "top-priority" attribute in json ("applyTopPriorityProperties()")